extension/builtin/default_registry.go

```diff
@@ -54,27 +53,45 @@ func (r *defaultRegistry) GetMapStorage() (storage.MapStorage, error) {
 	return mysql.NewMapStorage(r.db), nil
 }
 
-func (r *defaultRegistry) GetSigner(treeID int64) (*crypto.Signer, error) {
-	return r.signer, nil
+func (r *defaultRegistry) GetKeyProvider() (keys.Provider, error) {
```

Comment:
Go style doesn't typically use "Get" as a function prefix. Consider `SignerFactory()`

Btw, this is the right abstraction to be making here. Good call.